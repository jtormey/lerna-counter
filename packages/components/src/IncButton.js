import React from 'react'
import Button from './Button'

const IncButton = ({ onInc }) => (
  <Button title='increment' onClick={onInc} />
)

export default IncButton
