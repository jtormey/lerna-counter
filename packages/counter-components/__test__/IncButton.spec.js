import React from 'react'
import IncButton from '../src/IncButton'
import { render, mount } from 'enzyme'

describe('IncButton', () => {
  it('should render a button', () => {
    let wrapper = render(
      <IncButton />
    )
    expect(wrapper.toString()).toMatchSnapshot()
  })

  it('should trigger an increment', () => {
    let mockFn = jest.fn()
    let wrapper = mount(
      <IncButton onInc={mockFn} />
    )
    wrapper.find('button').simulate('click')
    expect(mockFn.mock.calls.length).toEqual(1)
  })
})
