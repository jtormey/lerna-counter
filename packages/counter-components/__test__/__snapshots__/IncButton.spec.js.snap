// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IncButton should render a button 1`] = `"<button>increment</button>"`;
